compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -170,13 +179,19 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
                 handleException {
                     it.onError()
                 }
+                if (hasInternalError) {
+                    return AnalysisResult.internalError(BindingContext.EMPTY, internalError)
+                }
             }
         } else {
             if (finished) {
                 processors.forEach {
                     handleException {
                         it.finish()
                     }
+                    if (hasInternalError) {
+                        return AnalysisResult.internalError(BindingContext.EMPTY, internalError)
+                    }
```

Comment:
`hasInternalError` can be saved by making `handleException` return. e.g.,
```
handleException {
    ...
}?.let { return it }
```

Reply:
This will result in a `return null` in handleException though, I am not sure if this is better.